### ISSUE TITLE
[idea] Open source waka time server implementation

### ISSUE THREAD

**Issue by at15** [dyweb/mos#7]
https://wakatime.com can be used to track how you spend your time for coding, however it only have three week history for free user. The server API is quite simple https://wakatime.com/developers. The storage should be a RDBMS + Time series database. Like MySQL + InfluxDB.  If search is needed, things like Elasticsearch can be introduced. Front end is mainly focus on dashboard, echart is enough. A rough plan could be like following

- [ ] choose the language and framework. I prefer language that does not require runtime or the runtime is easy to install on all platforms, so you can run the server locally, i.e. Golang, NodeJS, Python
- [ ] choose the database 
- [ ] implement basic API
- [ ] fork wakatime editor plugins to support pointing to custom server
- [ ] *allow team share and leaderboard
- [ ] *integration with GitLab

**Comment by gaocegege** [comment]
Good idea, but I'm afraid that there is no one interested in it now.
🤔  Because you know, this idea may be a little complicated, and I think there are little people to build a service like that. I would choose to pay for wakatime.

**Comment by at15** [comment]
@gaocegege I don't think you can afford a bunch of people using waka time, and you can do some hook and mining on data easily if you have full control of the statistics.

**Comment by at15** [comment]
I chose to pay for wakatime .... until I finished building a workable TSDB

**Comment by at15** [comment]
I think I can have try to do something similar when building reika https://github.com/at15/reika

**Comment by arrowrowe** [comment]
@at15 looking forward

**Comment by gaocegege** [comment]
Hope you could implement it :)

**Comment by at15** [comment]
@gaocegege where there is hope, there is disappointment 😄

**Comment by gaocegege** [comment]
进度如何了，婶婶

**Comment by at15** [comment]
@gaocegege 为了性能，我准备直接上 https://github.com/at15/gegecece 了，你怕不怕

**Comment by xplorld** [comment]
gaoce鸽鸽进度如何了鸭

**Comment by gaocegege** [comment]
It seems that @at15 is busy with his another wheel, this idea can be assigned to others.
